server/src/common/errors.ts

```diff
@@ -0,0 +1,13 @@
+export class NotFoundError extends Error {
```

Comment:
mmm I don't think it's worth splitting this kind of errors. Instead we can group it like
```
class ErrorResponseDTO extends Error {
message: string;
statusCode: number;
reason: any
 constructor(...) { ... }
}
```
not found could be `new ErrorResponseDTO(`The ${entity} with the given ID was not found`, StatusCodes.NOT_FOUND, 'if needed any detail')`

invalid parameter error  could be new ErrorResponseDTO(`Bad request`, StatusCodes.BAD_REQUEST, 'if needed any detail')